### DISCUSSION TITLE
Add an x-destroy directive called when the x-data node is removed from the DOM

### DISCUSSION THREAD

**Discussion by pomartel** [alpinejs/alpine#878] [3. Feature ideas]
I was wondering if it would be possible to have the equivalent of `x-init` but for when the component is removed from the DOM to do some clean up.

I wrote this Code Pen as a proof of concept of a rich text editor (Froala) that is initialized in the `x-init` and deleted in the `x-destroy`: https://codepen.io/pomartel/pen/GRqLpMG?editors=1010

**Comment by KevinBatdorf** [comment]
Were you able to clean up Froala using the approach in your demo? 

I think this would be a small PR actually. Might be worth adding it to get some feedback from Caleb on it. 

If not I can help you make a plugin for it.

**Reply by pomartel** [reply]
Thanks! I'll try to dig deeper into this on Friday. Might be a good way to get more familiar with alpine source code as well.

**Reply by SimoTod** [reply]
One thing to consider is that mutation observers are a bit heavy, if you send a PR, I'd suggest to integrate your code in the existing one rather than create a new observer.
